--- a/external/openglcts/modules/common/glcPackedDepthStencilTests.cpp
+++ b/external/openglcts/modules/common/glcPackedDepthStencilTests.cpp
@@ -382,6 +382,7 @@
 
 void BaseTest::init(void)
 {
+    m_defaultFBO = m_context.getRenderContext().getDefaultFramebuffer();
     if (glu::isContextTypeES(m_context.getRenderContext().getType()))
     {
         m_textureTargets              = esTexTargets;
